### PR TITLE
Update alignment of checkbox in table header

### LINKED ISSUE #1181: Fix padding of checkbox on Table header to align with checkboxes below it

### PR DESCRIPTION
**Pull request by taysea**
#### What does this PR do?
When `onSelect` is added as a prop for DataTable, a checkbox appears in the header cell. However, the alignment was off with the checkboxes in the body cells because we have removed padding from header cells. This adjusts the alignment by identifying the `label` which surrounds the checkbox and justifying it to the center.

#### Where should the reviewer start?
aries-site/src/themes/aries.js

#### What testing has been done on this PR?
Locally looking at DataTable w select.

#### How should this be manually tested?
Same as above.

#### What are the relevant issues?
Closes #1181 

#### Should this PR be mentioned in Design System updates?
No.

#### Is this change backwards compatible or is it a breaking change?
Yes.

### CODE REVIEW

File: aries-site/src/themes/aries.js [hunk 1]
```diff
@@ -89,6 +89,10 @@ export const aries = deepMerge(hpe, {
           height: 100%;
           justify-content: center;
         }
+        // align onSelect checkbox to center of header cell
```
Comment (ericsoderberghp) [inline aries-site/src/themes/aries.js:92]: Why isn't this in the HPE theme?

Comment (taysea) [reply aries-site/src/themes/aries.js:92]: we will be moving all of this over to the HPE theme soon as the dataTable theming is fluctuating less now. As designs were getting finalized we wanted to iron items out first before placing on theme branch.

**Review by halocline — COMMENTED** [review]
Unrelated to this work, but I am noticing that with the update of the yarn.lock we have visual regressions related to Menu (padding removed, disabled styling no longer applying, etc.):
Screen Shot 2020-10-12 at 12 13 32 PM

**Comment by taysea** [comment]
> Unrelated to this work, but I am noticing that with the update of the yarn.lock we have visual regressions related to Menu (padding removed, disabled styling no longer applying, etc.):
> Screen Shot 2020-10-12 at 12 13 32 PM

uh oh. good catch there. that is definitely something to look at ..

**Pull request merged**